sdk/storage/azure-storage-blob/src/main/java/com/azure/android/storage/blob/StorageBlobServiceImpl.java

```diff
@@ -1013,6 +1093,116 @@ public void onFailure(@NonNull Call<Void> call, @NonNull Throwable t) {
         }
     }
 
+    private BlobSetTagsResponse setBlobTagsWithRestResponseIntern(String containerName,
+                                                                  String blobName,
+                                                                  Integer timeout,
+                                                                  String versionId,
+                                                                  String iftags,
+                                                                  Map<String, String> tags,
+                                                                  String version,
+                                                                  String requestId,
+                                                                  CancellationToken cancellationToken,
+                                                                  CallbackWithHeader<Void, BlobSetTagsHeaders> callback) {
+
+        cancellationToken = cancellationToken == null ? CancellationToken.NONE : cancellationToken;
+
+        final String comp = "tags";
+
+        List<BlobTag> blobTagSet = new ArrayList<>(tags.size());
+        for (Map.Entry<String, String> entry : tags.entrySet()) {
+            blobTagSet.add(new BlobTag().setKey(entry.getKey()).setValue(entry.getValue()));
+        }
+        BlobTags blobTags = new BlobTags();
+        blobTags.setBlobTagSet(blobTagSet);
+
+        RequestBody tagsBody;
+        try {
+            tagsBody = RequestBody.create(MediaType.get("application/xml; charset=utf-8"),
+                serializerAdapter.serialize(blobTags, SerializerFormat.XML));
+        } catch (IOException ioe) {
+            if (callback != null) {
+                callback.onFailure(ioe, null);
+
+                return null;
+            } else {
+                throw new RuntimeException(ioe);
+            }
+        }
+
+        Call<ResponseBody> call = service.setBlobTags(containerName,
+            blobName,
+            timeout,
+            versionId,
+            null, // TODO: calculate Md5?
+            null,
+            iftags,
+            XMS_VERSION,
+            requestId,
+            tagsBody,
+            comp
+        );
+
+        ((CancellationTokenImpl) cancellationToken).registerOnCancel(() -> {
+            call.cancel();
+        });
+
+        if (callback != null) {
+            executeCall(call, new retrofit2.Callback<ResponseBody>() {
+                @Override
+                public void onResponse(@NonNull Call<ResponseBody> call, @NonNull Response<ResponseBody> response) {
+                    if (response.isSuccessful()) {
+                        if (response.code() == 200) {
```

Comment:
Seems as per swagger, valid response code is [204](https://github.com/Azure/azure-sdk-for-java/blob/9de1e77a1329ad9291e261f737dd7db00209d6d6/sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/implementation/BlobsImpl.java#L219)

Reply:
Nice catch :)